skfda/ml/classification/_centroid_classifiers.py

```diff
@@ -0,0 +1,216 @@
+
+        return self
+
+    def predict(self, X):
+        """Predict the class labels for the provided data.
+
+        Args:
+            X (:class:`FDataGrid`): FDataGrid with the test samples.
+
+        Returns:
+            y (np.array): array of shape [n_samples] or
+            [n_samples, n_outputs] with class labels for each data sample.
+        """
+        sklearn_check_is_fitted(self)
+
+        return self.classes_[pairwise_distance(self.metric)(
+            X,
+            self.centroids_,
+        ).argmin(axis=1)
+        ]
+
+
+class DTMClassifier(BaseEstimator, ClassifierMixin):
+    """Distance to trimmed means (DTM) classification.
+
+    Test samples are classified to the class that minimizes the distance of
+    the observation to the trimmed mean of the group.
+
+    Parameters:
+        proportiontocut (float): indicates the percentage of functions to
+            remove. It is not easy to determine as it varies from dataset to
+            dataset.
+        depth_method (Depth, default
+            :class:`ModifiedBandDepth <skfda.depth.ModifiedBandDepth>`):
+            The depth class used to order the data. See the documentation of
+            the depths module for a list of available depths. By default it
+            is ModifiedBandDepth.
+        metric (Callable, default
+            :func:`lp_distance <skfda.misc.metrics.lp_distance>`):
+            Distance function between two functional objects. See the
+            documentation of the metrics module for a list of available
+            metrics.
+
+    Examples:
+        Firstly, we will import and split the Berkeley Growth Study dataset
+
+        >>> from skfda.datasets import fetch_growth
+        >>> from sklearn.model_selection import train_test_split
+        >>> dataset = fetch_growth()
+        >>> fd = dataset['data']
+        >>> y = dataset['target']
+        >>> X_train, X_test, y_train, y_test = train_test_split(
+        ...     fd, y, test_size=0.25, stratify=y, random_state=0)
+
+        We will fit a Distance to trimmed means classifier
+
+        >>> from skfda.ml.classification import DTMClassifier
+        >>> clf = DTMClassifier(proportiontocut=0.25)
+        >>> clf.fit(X_train, y_train)
+        DTMClassifier(...)
+
+        We can predict the class of new samples
+
+        >>> clf.predict(X_test) # Predict labels for test samples
+        array([1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 1,
+                1, 0, 0, 0, 0, 1, 1, 1, 1, 0, 1, 1])
+
+        Finally, we calculate the mean accuracy for the test data
+
+        >>> clf.score(X_test, y_test)
+        0.875
+
+    See also:
+        :class:`~skfda.ml.classification.MaximumDepthClassifier`
+
+    References:
+        Fraiman, R. and Muniz, G. (2001). Trimmed means for functional
+        data. Test, 10, 419-440.
+    """
+
+    def __init__(
+        self,
+        proportiontocut: float,
+        depth_method: Depth = None,
+        metric: Callable = lp_distance,
+    ) -> None:
```

Comment:
Deberíamos poner el strictness de darglint a "long" para que nos permita tener docstrings cortos sin especificar argumentos, return, etc, pero que compruebe los argumentos si los ponemos.

Reply:
Yes. However, I think that in this case, it would be better to not have docstrings in `__init__` methods. Classes docstrings all ready document the parameters and moreover this is consistent with Google`s style guide.

Reply:
I agree. It is consistent with scikit-learn too.

Reply:
Whoops, I switched languages before 😅.